java_src/foam/dao/history/HistoryDAO.java

```diff
@@ -70,9 +69,8 @@ public FObject put(FObject obj) {
     FObject current = this.find(obj);
 
     // add new history record
-    String objectId = (String) ((PropertyInfo) obj.getClassInfo().getAxiomByName("objectId")).f(obj);
     HistoryRecord historyRecord = new HistoryRecord();
-    historyRecord.setObjectId(objectId);
+    historyRecord.setObjectId(obj.getClassInfo().getId());
```

Comment:
I was wrong, that's the id of the ClassInfo, not the object.
Your original code was right, except that it should have been 'id' instead of 'objectId'.